feat(stats): add nanmgmean package

hans2001 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5576 

## Description

> What is the purpose of this pull request?

This pull request introduces the **nanmgmean** package—a thin wrapper around **mgmean** that enhances robustness. It adds type checking so that if a NaN or non-numeric value is provided, the update is skipped and the accumulator continues to return the previously computed mean directional accuracy. Test cases have been updated to verify that these edge cases are correctly handled.

This pull request:

test coverage: 
<img width="644" alt="Screenshot 2025-03-20 at 10 51 48 PM" src="https://github.com/user-attachments/assets/1e5f010b-de53-404c-9cbb-ac4bafbbd57c" />

example:
<img width="163" alt="Screenshot 2025-03-20 at 10 52 21 PM" src="https://github.com/user-attachments/assets/ed6ca854-5ad0-4f30-9b6e-111047ef63e7" />

benchmark: 
<img width="307" alt="Screenshot 2025-03-20 at 10 52 05 PM" src="https://github.com/user-attachments/assets/75b122f0-ec3f-42cb-9f2a-2669449bf6c5" />

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5576 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

hans2001 committed:
feat(stats): add nanmgmean package

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

hans2001 committed:
fix(readme): udpate example

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/benchmark/benchmark.js:
```diff
@@ -0,0 +1,69 @@
+	var i;
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		f = incrnanmgmean( (i%5)+1 );
```
```suggestion
		f = incrnanmgmean( ( i%5 )+1 );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+type accumulator = ( x?: number ) => number | null;
+
+/**
+* Returns an accumulator function which incrementally computes a moving geometric mean.
```
```suggestion
* Returns an accumulator function which incrementally computes a moving geometric mean, ignoring `NaN` value.
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+* @returns accumulator function
+*
+* @example
+* var accumulator = incrmgmean( 3 );
```
```suggestion
* var accumulator = incrnanmgmean( 3 );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+* v = accumulator();
+* // returns ~4.22
+*/
+declare function incrmgmean( W: number ): accumulator;
```
```suggestion
declare function incrnanmgmean( W: number ): accumulator;
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+
+// EXPORTS //
+
+export = incrmgmean;
```
```suggestion
export = incrnanmgmean;
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,74 @@
+
+/**
+* If provided a value, returns an updated geometric mean; otherwise, returns the current geometric mean.
+*
```
Remove line from L25-29

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/docs/types/test.ts:
```diff
@@ -0,0 +1,66 @@
+* limitations under the License.
+*/
+
+import incrmgmean = require( './index' );
```
Update `incrmgmean` to `incrnanmgmean` in whole file.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmgmean/package.json:
```diff
@@ -0,0 +1,75 @@
+{
+  "name": "@stdlib/stats/incr/nanmgmean",
+  "version": "0.0.0",
+  "description": "Compute a moving geometric mean incrementally.",
```
```suggestion
  "description": "Compute a moving geometric mean incrementally, ignoring `NaN` value.",
```

hans2001 committed:
fix(types): update careless mistakes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

hans2001 commented:
@hrshya i have updated the corresponding portions based on your comments, Thanks for the reminder!

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/stats_incr_nanmgmean

Planeshifter committed:
chore: update copyright years

Planeshifter committed:
fix: address additional review feedback

Planeshifter reviewed (approved):
LGTM after clean-up.
